### PR TITLE
Fix typo in CMakeList.

### LINKED ISSUE #12237: Potential bug in the cmake files.
> Description: Some variables are not matched in cmake files.  
e.g: 
1. FindNUMA.cmake:  line 19-21 marked_as_advanced(NUMA_LIBRARIES, NUMA_INCLUDE_DIRS), while in the CMakeLists.txt. line 374, use include_directories(${NUMA_INCLUDE_DIR}). which is not correct. 
2. Findzstd.cmake line 21,  CMakeLists.txt line 169.

Although cmake will correctly load the header in the default search path, but it makes people confused.

### Expected behavior
CMakeLists.txt: line 374
 include_directories(${NUMA_INCLUDE_DIRS})
### Actual behavior
include_directories(${NUMA_INCLUDE_DIR})

### PR DESCRIPTION
**Pull request by rhubner**
Fix #12237

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -166,7 +166,7 @@ else()
   if(WITH_ZSTD)
     find_package(zstd REQUIRED)
     add_definitions(-DZSTD)
-    include_directories(${ZSTD_INCLUDE_DIR})
+    include_directories(${ZSTD_INCLUDE_DIRS})
```
Comment (adamretter) [inline CMakeLists.txt:169]: I think this also needs fixing in `cmake/modules/Findzstd.cmake` where it seems to be in mixed-case.

**Review by adamretter — CHANGES_REQUESTED** [review]
I think a small fix is still needed in `Findzstd.cmake`.
It might also be worth checking the other files in `cmake/modules/` as well?

**Review by adamretter — APPROVED** [review]
Now LGTM. Thanks @rhubner

**Comment by adamretter** [comment]
@ajkr @pdillinger Can we get this one imported please?

**Review by ajkr — APPROVED** [review]

**Comment by georgthegreat** [comment]
This PR broke nixpkgs build. It looks like their version of zstd does not export ZSTD_INCLUDE_DIR

**Comment by georgthegreat** [comment]
We have an internal system built atop of nixpkgs.
I am receiving the following error from it:

```
-- Found assembler: /nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/gcc
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found ZLIB: /nix/store/8xgb8phqmfn9h971q7dg369h647i1aa0-zlib-1.3/lib/libz.so (found version "1.3")
-- Found BZip2: /nix/store/155qsyx1mv11fsi48nz4dlc0vh1a3drx-bzip2-1.0.8/lib/libbz2.so (found version "1.0.8")
-- Looking for BZ2_bzCompressInit
-- Looking for BZ2_bzCompressInit - found
-- Found lz4: /nix/store/874gxh0sgwkbwb7nfbp72sn7a058b7sa-lz4-1.9.4/lib/liblz4.so
CMake Error at /nix/store/vnhl4zdy7igx9gd3q1d548vwzz15a9ma-cmake-3.27.7/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find zstd (missing: ZSTD_INCLUDE_DIRS)
Call Stack (most recent call first):
  /nix/store/vnhl4zdy7igx9gd3q1d548vwzz15a9ma-cmake-3.27.7/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/Findzstd.cmake:17 (find_package_handle_standard_args)
  CMakeLists.txt:167 (find_package)
```

I have worked it around by adding the following:
```
  cmakeFlags = attrs.cmakeFlags ++ [
    # Configuration fails otherwise as ZSTD_INCLUDE_DIRS is checked instead of ZSTD_INCLUDE_DIR
    "-DZSTD_INCLUDE_DIRS=${zstd.dev}"
  ];
```

I am not quite familiar with cmake internals to debug this.

**Comment by rhubner** [comment]
Hello @georgthegreat and sorry for late reply,

Unfortunately I don't know nixos, but I can confirm that we recently changed the CMake build script. The refactor follows the pattern which we used to specify include dir for other libraries. Your fix with `"-DZSTD_INCLUDE_DIRS=${zstd.dev}"` makes sense and sounds legit for me. What I don't understand, how did you build it before without providing libzstd include path.

About CMake. During the build script generation with CMake, you can specify path where you have your libzstd. This allows you to use your version of zstd and not for example system/distribution version. I assume that when you are building nixos, you also build libzstd and then CMake needs to know where it is, if it's not in the default search path.

Radek

**Comment by georgthegreat** [comment]
I have tried to understand why previous version worked, but I was unable to get to the root of it.
It looks like CMake's find_package does not provide any include-related variables out of the box, so it is up to the specific package manager to decide whether to provide this variables and how to name them.

I agree that `-DZSTD_INCLUDE_DIRS` solution solves the problems.
I am fine with keeping it.

**Pull request closed without merging**